lte/gateway/python/magma/mobilityd/ip_address_man.py

```diff
@@ -133,70 +125,18 @@ def __init__(self,
-            self._assigned_ip_blocks = set()  # {ip_block}
-            self.sid_ips_map = defaultdict(IPDesc)  # {SID=>IPDesc}
-            self._dhcp_gw_info = UplinkGatewayInfo(defaultdict(str))
-            self._dhcp_store = {}  # mac => DHCP_State
-        else:
-            if not redis_port:
-                raise ValueError(
-                    'Must specify a redis_port in mobilityd config.')
-            client = get_default_client()
-            self._assigned_ip_blocks = store.AssignedIpBlocksSet(client)
-            self.sid_ips_map = store.IPDescDict(client)
-            self._dhcp_gw_info = UplinkGatewayInfo(store.GatewayInfoMap())
-            self._dhcp_store = store.MacToIP()  # mac => DHCP_State
-
-        self.ip_state_map = IpDescriptorMap(persist_to_redis, redis_port)
-        logging.info("Using allocator: %s static ip: %s multi_apn %s",
-                     self.allocator_type,
-                     self.static_ip_enabled,
-                     self.multi_apn)
-
-        if self.allocator_type == MobilityD.IP_POOL:
-            self._dhcp_gw_info.read_default_gw()
-            ip_allocator = IpAllocatorPool(self._assigned_ip_blocks,
-                                           self.ip_state_map,
-                                           self.sid_ips_map)
-        elif self.allocator_type == MobilityD.DHCP:
-            iface = config.get('dhcp_iface', 'dhcp0')
-            retry_limit = config.get('retry_limit', 300)
-            ip_allocator = IPAllocatorDHCP(assigned_ip_blocks=self._assigned_ip_blocks,
-                                           ip_state_map=self.ip_state_map,
-                                           iface=iface,
-                                           retry_limit=retry_limit,
-                                           dhcp_store=self._dhcp_store,
-                                           gw_info=self._dhcp_gw_info)
-        else:
-            raise ValueError("Unknown IP allocator type: %s" % self.allocator_type)
-
-        if self.static_ip_enabled:
-            ip_allocator = IPAllocatorStaticWrapper(subscriberdb_rpc_stub=subscriberdb_rpc_stub,
-                                                    ip_allocator=ip_allocator,
-                                                    gw_info=self._dhcp_gw_info,
-                                                    assigned_ip_blocks=self._assigned_ip_blocks,
-                                                    ip_state_map=self.ip_state_map)
-
-        if self.multi_apn:
-            self.ip_allocator = IPAllocatorMultiAPNWrapper(subscriberdb_rpc_stub=subscriberdb_rpc_stub,
-                                                           ip_allocator=ip_allocator)
-        else:
-            self.ip_allocator = ip_allocator
+        # Load store dependencies
+        self._ip_state_map = store.ip_state_map
+        self._sid_ips_map = store.sid_ips_map
+        self._dhcp_gw_info = store.dhcp_gw_info
+        self._dhcp_store = store.dhcp_store
```

Comment:
just track the store and update callsties of the storage maps to use `store` instead (e.g. self._ip_state_map[foo] -_ self.store.ip_state_map[foo])